mailbox/elasticsearch/src/main/java/org/apache/james/mailbox/elasticsearch/search/ElasticSearchSearcher.java

```diff
@@ -70,26 +70,19 @@ public ElasticSearchSearcher(Client client, QueryConverter queryConverter, int s
         this.mailboxIdFactory = mailboxIdFactory;
         this.messageIdFactory = messageIdFactory;
     }
-    
-    public Stream<MessageSearchIndex.SearchResult> search(
-        Collection<MailboxId> mailboxIds,
-        SearchQuery query,
-        Optional<Long> limit
-    ) throws MailboxException {
+
+    public Stream<MessageSearchIndex.SearchResult> search(Collection<MailboxId> mailboxIds, SearchQuery query,
```

Comment:
I prefer my way but thats ok we can keep your way

Reply:
Well, your way is:

 - 1/ not standard. Let's keep standard way to deal with methods with too much parameters
 - 2/ I can hardly read it.

Reply:
1/ what is the standard (link) and who define it

2/ Me it's the contrary you way work well in this case but doesn't scale when we have more than 6 arguments and more than two exceptions..

Reply:
1/ No link, it's team practice. Open any class, you'll see we always indent several level further methods arguments (IE CassandraMailboxSessionMapperFactory SetMessagesCreationProcessor, etc... )
2/ Should we have more than 6 args?

Reply:
2/ this is often the case on some constructor unfortunately.